dyninstAPI_RT build regression on power

@kupsch It looks like #1246 is causing some types to not get included from POSIX headers on power. This was on llnl.cs.rice.edu which has glibc 2.27 and gcc 8.3.0.

```
dyninstAPI_RT/src/RTheap.c: In function 'DYNINSTos_malloc':
dyninstAPI_RT/src/RTheap.c:113:28: warning: implicit declaration of function 'getpagesize' [-Wimplicit-function-declaration]
   if (psize == -1) psize = getpagesize();
                            ^~~~~~~~~~~
dyninstAPI_RT/src/RTheap-linux.c:65:33: error: 'MAP_ANONYMOUS' undeclared here (not in a function)
 int     DYNINSTheap_mmapFlags = MAP_ANONYMOUS | MAP_PRIVATE;
                                 ^~~~~~~~~~~~~
make[5]: *** [CMakeFiles/dyninstAPI_RT.dir/src/RTheap-linux.c.o] Error 1
make[5]: *** Waiting for unfinished jobs....
dyninstAPI_RT/src/RTlinux.c:65:8: error: unknown type name 'pthread_t'
 extern pthread_t pthread_self(void);
        ^~~~~~~~~
dyninstAPI_RT/src/RTlinux.c: In function 't_kill':
dyninstAPI_RT/src/RTlinux.c:96:18: warning: implicit declaration of function 'syscall'; did you mean 'strcoll'? [-Wimplicit-function-declaration]
         result = syscall(SYS_tkill, pid, sig);
                  ^~~~~~~
                  strcoll
dyninstAPI_RT/src/RTlinux.c:102:18: warning: implicit declaration of function 'kill'; did you mean 't_kill'? [-Wimplicit-function-declaration]
         result = kill(pid, sig);
                  ^~~~
                  t_kill
dyninstAPI_RT/src/RTposix.c: In function 'DYNINSTasyncConnect':
dyninstAPI_RT/src/RTposix.c:139:5: error: 'P_tmpdir' undeclared (first use in this function); did you mean 'rmdir'?
     P_tmpdir, passwd_info->pw_name, pid, mutatee_pid);
     ^~~~~~~~
     rmdir
dyninstAPI_RT/src/RTposix.c:139:5: note: each undeclared identifier is reported only once for each function it appears in
dyninstAPI_RT/src/RTlinux.c: In function 'dyn_pthread_self':
dyninstAPI_RT/src/RTlinux.c:297:14: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
       return (dyntid_t) pthread_self();
              ^
dyninstAPI_RT/src/RTposix.c: At top level:
dyninstAPI_RT/src/RTposix.c:267:41: error: unknown type name 'siginfo_t'; did you mean 'ino_t'?
 extern void dyninstTrapHandler(int sig, siginfo_t *info, void *context);
                                         ^~~~~~~~~
                                         ino_t
dyninstAPI_RT/src/RTposix.c: In function 'DYNINSTinitializeTrapHandler':
dyninstAPI_RT/src/RTposix.c:272:21: error: storage size of 'new_handler' isn't known
    struct sigaction new_handler;
                     ^~~~~~~~~~~
dyninstAPI_RT/src/RTposix.c:283:31: error: 'dyninstTrapHandler' undeclared (first use in this function); did you mean 'dyninstTrapTranslate'?
    new_handler.sa_sigaction = dyninstTrapHandler;
                               ^~~~~~~~~~~~~~~~~~
                               dyninstTrapTranslate
dyninstAPI_RT/src/RTposix.c:285:4: warning: implicit declaration of function 'sigemptyset' [-Wimplicit-function-declaration]
    sigemptyset(&new_handler.sa_mask);
    ^~~~~~~~~~~
dyninstAPI_RT/src/RTlinux.c: In function 't_kill':
dyninstAPI_RT/src/RTposix.c: In function 'DYNINSTasyncConnect':
dyninstAPI_RT/src/RTlinux.c:96:18: warning: implicit declaration of function 'syscall'; did you mean 'strcoll'? [-Wimplicit-function-declaration]
         result = syscall(SYS_tkill, pid, sig);
                  ^~~~~~~
                  strcoll
dyninstAPI_RT/src/RTposix.c:139:5: error: 'P_tmpdir' undeclared (first use in this function); did you mean 'rmdir'?
     P_tmpdir, passwd_info->pw_name, pid, mutatee_pid);
     ^~~~~~~~
     rmdir
dyninstAPI_RT/src/RTposix.c:139:5: note: each undeclared identifier is reported only once for each function it appears in
dyninstAPI_RT/src/RTlinux.c:102:18: warning: implicit declaration of function 'kill'; did you mean 't_kill'? [-Wimplicit-function-declaration]
         result = kill(pid, sig);
                  ^~~~
                  t_kill
dyninstAPI_RT/src/RTposix.c:286:27: error: 'SA_SIGINFO' undeclared (first use in this function); did you mean 'SO_MEMINFO'?
    new_handler.sa_flags = SA_SIGINFO | SA_NODEFER;
                           ^~~~~~~~~~
                           SO_MEMINFO
dyninstAPI_RT/src/RTposix.c: At top level:
dyninstAPI_RT/src/RTposix.c:267:41: error: unknown type name 'siginfo_t'; did you mean 'ino_t'?
 extern void dyninstTrapHandler(int sig, siginfo_t *info, void *context);
                                         ^~~~~~~~~
                                         ino_t
dyninstAPI_RT/src/RTposix.c: In function 'DYNINSTinitializeTrapHandler':
dyninstAPI_RT/src/RTposix.c:272:21: error: storage size of 'new_handler' isn't known
    struct sigaction new_handler;
                     ^~~~~~~~~~~
In file included from /usr/include/ucontext.h:26,
                 from dyninstAPI_RT/src/RTlinux.c:321:
/usr/include/sys/ucontext.h: At top level:
/usr/include/sys/ucontext.h:138:5: error: unknown type name 'stack_t'
     stack_t uc_stack;
     ^~~~~~~
dyninstAPI_RT/src/RTposix.c:286:40: error: 'SA_NODEFER' undeclared (first use in this function); did you mean 'SO_NOFCS'?
    new_handler.sa_flags = SA_SIGINFO | SA_NODEFER;
                                        ^~~~~~~~~~
                                        SO_NOFCS
/usr/include/sys/ucontext.h:171:5: error: unknown type name 'sigset_t'
     sigset_t    uc_sigmask;
     ^~~~~~~~
dyninstAPI_RT/src/RTposix.c:283:31: error: 'dyninstTrapHandler' undeclared (first use in this function); did you mean 'dyninstTrapTranslate'?
    new_handler.sa_sigaction = dyninstTrapHandler;
                               ^~~~~~~~~~~~~~~~~~
                               dyninstTrapTranslate
dyninstAPI_RT/src/RTposix.c:288:13: warning: implicit declaration of function 'sigaction' [-Wimplicit-function-declaration]
    result = sigaction(signo, &new_handler, NULL);
             ^~~~~~~~~
dyninstAPI_RT/src/RTposix.c:272:21: warning: unused variable 'new_handler' [-Wunused-variable]
    struct sigaction new_handler;
                     ^~~~~~~~~~~
dyninstAPI_RT/src/RTposix.c:285:4: warning: implicit declaration of function 'sigemptyset' [-Wimplicit-function-declaration]
    sigemptyset(&new_handler.sa_mask);
    ^~~~~~~~~~~
dyninstAPI_RT/src/RTlinux.c:373:34: error: unknown type name 'siginfo_t'; did you mean 'ino_t'?
 void dyninstTrapHandler(int sig, siginfo_t *sg, ucontext_t *context)
                                  ^~~~~~~~~
                                  ino_t
dyninstAPI_RT/src/RTposix.c:286:27: error: 'SA_SIGINFO' undeclared (first use in this function); did you mean 'SO_MEMINFO'?
    new_handler.sa_flags = SA_SIGINFO | SA_NODEFER;
                           ^~~~~~~~~~
                           SO_MEMINFO
make[5]: *** [CMakeFiles/dyninstAPI_RT_static.dir/src/RTposix.c.o] Error 1
make[5]: *** Waiting for unfinished jobs....
dyninstAPI_RT/src/RTposix.c:286:40: error: 'SA_NODEFER' undeclared (first use in this function); did you mean 'SO_NOFCS'?
    new_handler.sa_flags = SA_SIGINFO | SA_NODEFER;
                                        ^~~~~~~~~~
                                        SO_NOFCS
dyninstAPI_RT/src/RTposix.c:288:13: warning: implicit declaration of function 'sigaction' [-Wimplicit-function-declaration]
    result = sigaction(signo, &new_handler, NULL);
             ^~~~~~~~~
dyninstAPI_RT/src/RTposix.c:272:21: warning: unused variable 'new_handler' [-Wunused-variable]
    struct sigaction new_handler;
                     ^~~~~~~~~~~
In file included from /usr/include/ucontext.h:26,
                 from dyninstAPI_RT/src/RTlinux.c:321:
/usr/include/sys/ucontext.h: At top level:
dyninstAPI_RT/src/RTlinux.c:444:36: warning: 'getStaticTrapMap' defined but not used [-Wunused-function]
 static struct trap_mapping_header *getStaticTrapMap(unsigned long addr)
                                    ^~~~~~~~~~~~~~~~
/usr/include/sys/ucontext.h:138:5: error: unknown type name 'stack_t'
     stack_t uc_stack;
     ^~~~~~~
/usr/include/sys/ucontext.h:171:5: error: unknown type name 'sigset_t'
     sigset_t    uc_sigmask;
     ^~~~~~~~
make[5]: *** [CMakeFiles/dyninstAPI_RT_static.dir/src/RTlinux.c.o] Error 1
make[5]: *** [CMakeFiles/dyninstAPI_RT.dir/src/RTposix.c.o] Error 1
dyninstAPI_RT/src/RTlinux.c:373:34: error: unknown type name 'siginfo_t'; did you mean 'ino_t'?
 void dyninstTrapHandler(int sig, siginfo_t *sg, ucontext_t *context)
                                  ^~~~~~~~~
                                  ino_t
dyninstAPI_RT/src/RTlinux.c:444:36: warning: 'getStaticTrapMap' defined but not used [-Wunused-function]
 static struct trap_mapping_header *getStaticTrapMap(unsigned long addr)
                                    ^~~~~~~~~~~~~~~~
make[5]: *** [CMakeFiles/dyninstAPI_RT.dir/src/RTlinux.c.o] Error 1
dyninstAPI_RT/src/RTheap.c: In function 'DYNINSTos_malloc':
dyninstAPI_RT/src/RTheap.c:113:28: warning: implicit declaration of function 'getpagesize' [-Wimplicit-function-declaration]
   if (psize == -1) psize = getpagesize();
                            ^~~~~~~~~~~
make[4]: *** [CMakeFiles/dyninstAPI_RT.dir/all] Error 2
make[4]: *** Waiting for unfinished jobs....
make[4]: *** [CMakeFiles/dyninstAPI_RT_static.dir/all] Error 2
make[3]: *** [all] Error 2
make[2]: *** [CMakeFiles/DyninstRT] Error 2
make[1]: *** [CMakeFiles/DyninstRT.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make: *** [all] Error 2
```

--- a/dyninstAPI_RT/src/RTheap-linux.c
+++ b/dyninstAPI_RT/src/RTheap-linux.c
@@ -31,7 +31,7 @@
 /* $Id: RTheap-linux.c,v 1.9 2008/01/31 18:01:54 legendre Exp $ */
 /* RTheap-linux.c: Linux-specific heap components */
 
-#define _DEFAULT_SOURCE
+#define _GNU_SOURCE
 
 #include <stdlib.h>
 #include <stdio.h>

--- a/dyninstAPI_RT/src/RTheap.c
+++ b/dyninstAPI_RT/src/RTheap.c
@@ -31,7 +31,7 @@
 /* $Id: RTheap.c,v 1.25 2006/05/03 00:31:25 jodom Exp $ */
 /* RTheap.c: platform-generic heap management */
 
-#define _DEFAULT_SOURCE
+#define _GNU_SOURCE
 
 #include <stdlib.h>
 #include <stdio.h>

--- a/dyninstAPI_RT/src/RTlinux.c
+++ b/dyninstAPI_RT/src/RTlinux.c
@@ -33,7 +33,7 @@
  * RTlinux.c: mutatee-side library function specific to Linux
  ************************************************************************/
 
-#define _DEFAULT_SOURCE
+#define _GNU_SOURCE
 
 #include "dyninstAPI_RT/h/dyninstAPI_RT.h"
 #include "dyninstAPI_RT/src/RTthread.h"

--- a/dyninstAPI_RT/src/RTposix.c
+++ b/dyninstAPI_RT/src/RTposix.c
@@ -33,7 +33,7 @@
  * RTposix.c: runtime instrumentation functions for generic posix.
  ************************************************************************/
 
-#define _DEFAULT_SOURCE
+#define _GNU_SOURCE
 
 #include <stdio.h>
 #include <stdlib.h>

--- a/dyninstAPI_RT/src/RTsignal.c
+++ b/dyninstAPI_RT/src/RTsignal.c
@@ -32,7 +32,7 @@
  * RTsignal.c: C-language signal handling code
  ************************************************************************/
 
-#define _DEFAULT_SOURCE
+#define _GNU_SOURCE
 
 #include <stdio.h>
 #include <signal.h>

